lib/ansible/modules/net_tools/nmcli.py

```diff
@@ -1102,7 +1102,7 @@ def main():
             conn_name=dict(required=True, type='str'),
             master=dict(required=False, default=None, type='str'),
             ifname=dict(required=False, default=None, type='str'),
-            type=dict(required=False, default=None, choices=['ethernet', 'team', 'team-slave', 'bond', 'bond-slave', 'bridge', 'vlan'], type='str'),
+            type=dict(required=True, default=None, choices=['ethernet', 'team', 'team-slave', 'bond', 'bond-slave', 'bridge', 'vlan'], type='str'),
```

Comment:
This will have detrimental effects on operations like 'modify' and 'delete', which does not require connection type of 'connection' or 'network device'. Also, it will break backward compatibility.